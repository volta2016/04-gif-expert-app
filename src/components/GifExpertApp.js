import React, { useState } from 'react';
import  AddCategory from './AddCategory'
import GifGrid from './GifGrid';
import Header from './Header'
import Footer from './Footer';




const GifExpertApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState(['One Punch']);
    // const [categories, setCategories] = useState(defaultCategories);

    return (
      <>
        <Header />
        <div className="container">
          <h2>Find your favorite gif ðŸ˜„</h2>
          < AddCategory setCategories={ setCategories }/>
          <hr></hr>
          
          <ol>
            {
              categories.map(category => ( 
                    <GifGrid 
                      key={ category }
                      category={ category }
                    />
                ))
            }
          </ol>
        </div>
        <Footer />
      </>
    );
}

export default GifExpertApp;



