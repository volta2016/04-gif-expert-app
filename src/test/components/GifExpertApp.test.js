
import React from 'react'
import GifExpertApp from '../../components/GifExpertApp';
import {shallow} from 'enzyme';





describe('pruebas en componente <GifExpertApp />', () => {
  test('debe de mostrarte el componente corectamente', () => {
    
    const wrapper  = shallow(<GifExpertApp  />)
    expect(wrapper).toMatchSnapshot();
  })
  
});
