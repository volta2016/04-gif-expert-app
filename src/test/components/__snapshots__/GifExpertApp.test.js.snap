// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en componente <GifExpertApp /> debe de mostrarte el componente corectamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
